### PR TITLE
Inject task factory

### PR DESCRIPTION
**Pull request by XiXiaPdx**
Do not merge until this merged and then rebase: https://github.com/newrelic/newrelic-gradle-verify-instrumentation/pull/24

The problem is ProjectTaskFactory is being instantiated inside the class to be tested, so it is difficult to test. Now, it is being injected. The first commit injects but removes `excludedVersions` from constructor (fixed with second commit).
 
The second commit is adding a `setExcludedVersions` method to ProjectTaskFactory. This mimics the pattern of `setPassesFile`.

### CODE REVIEW

File: src/main/java/com/newrelic/agent/instrumentation/verify/VerificationPlugin.java [hunk 1]
```diff
@@ -39,7 +39,7 @@ public void apply(Project target) {
                 verifyInstrumentation,
                 verifyCapstoneTask,
                 target.getLogger(),
-                destinationDir));
+                destinationDir, new ProjectTaskFactory(target, target.getLogger(), destinationDir)));
```
Comment (jeffalder) [inline src/main/java/com/newrelic/agent/instrumentation/verify/VerificationPlugin.java:42]:
```suggestion
                destinationDir, 
                new ProjectTaskFactory(target, target.getLogger(), destinationDir)));
```

Consistency in formatting

File: src/test/groovy/com/newrelic/agent/instrumentation/verify/VerifierTest.groovy [hunk 2]
```diff
@@ -28,6 +51,9 @@ class VerifierTest {
 
     @Test
     void testAddDependenciesForVerifierTasks() {
+        MockitoAnnotations.initMocks(this);
```
Comment (jeffalder) [inline src/test/groovy/com/newrelic/agent/instrumentation/verify/VerifierTest.groovy:54]: This should be added to a `@Before` method.

File: src/test/java/com/newrelic/agent/instrumentation/verify/ProjectTaskFactoryTest.java [hunk 3]
```diff
@@ -229,13 +222,16 @@ private void givenProjectIsConfigured(ProjectTweaks... tweaks) {
     }
 
     private void givenTaskFactoryExcludingVersionTwo() {
-        target = new ProjectTaskFactory(project, Collections.singletonList("foo:bar:2.0"), NOPLogger.NOP_LOGGER, tempDir.toFile());
+        target = new ProjectTaskFactory(project, NOPLogger.NOP_LOGGER, tempDir.toFile());
         target.setPassesFile(verifyInstrumentationOptions.passesFileName);
+        Set<String> excludedVersion = new HashSet<>(Arrays.asList("foo:bar:2.0"));
```
Comment (jeffalder) [inline src/test/java/com/newrelic/agent/instrumentation/verify/ProjectTaskFactoryTest.java:227]:
```suggestion
        Set<String> excludedVersions = Collections.singleton("foo:bar:2.0");
```

**Comment by XiXiaPdx** [comment]
Ok, made all the suggested changes.

**Pull request closed without merging**